src/smc-webapp/project_store.coffee

```diff
@@ -1258,6 +1127,159 @@ class ProjectStore extends Store
 
         return x
 
+    convert_sagenb_worksheet: (filename, cb) =>
+        async.series([
+            (cb) =>
+                ext = misc.filename_extension(filename)
+                if ext == "sws"
+                    cb()
+                else
+                    i = filename.length - ext.length
+                    new_filename = filename.slice(0, i-1) + ext.slice(3) + '.sws'
+                    salvus_client.exec
+                        project_id : @project_id
+                        command    : "cp"
+                        args       : [filename, new_filename]
+                        cb         : (err, output) =>
+                            if err
+                                cb(err)
+                            else
+                                filename = new_filename
+                                cb()
+            (cb) =>
+                salvus_client.exec
+                    project_id : @project_id
+                    command    : "smc-sws2sagews"
+                    args       : [filename]
+                    cb         : (err, output) =>
+                        cb(err)
+        ], (err) =>
+            if err
+                cb(err)
+            else
+                cb(undefined, filename.slice(0,filename.length-3) + 'sagews')
+        )
+
+    convert_docx_file: (filename, cb) =>
```

Comment:
ACTIONS: 
  - they change state -- functions that do things and modify the store.  Do NOT return values.

STORE: 
  - stores state and has get'ers that return values.  These functions do NOT change state.

You just moved a bunch of functioniality to the store, but these are definitely not methods that belong in the store.  They could be actions if they do something, then store a result in the store.    However, these seem to just be traditional async callback functions.  There is no rule that **everything** needs to be an action or in the store.    In fact, there are rules on how the actions and stores work and what sort of functions they have, so we can easily reason about them.